oktapam/resource_database.go

```diff
@@ -0,0 +1,380 @@
+		},
+		attributes.BasicAuth: {
+			Type: schema.TypeList,
+			// If any new auth methods are created, we should utilize ConflictsWith to ensure only one is set at a time.
+			//ConflictsWith: []string{parent_block_name.0.child_attribute_name},
+			Required:    true,
+			MinItems:    1,
+			MaxItems:    1,
+			Description: descriptions.DatabaseAuthDetails,
+			Elem:        mySQLBasicAuthDetails,
+		},
+	},
+}
+
+var mySQLBasicAuthDetails = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		attributes.Username: {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: descriptions.UserName,
+		},
+		attributes.Password: {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: descriptions.DatabasePassword,
+			Sensitive:   true,
+		},
+		attributes.Secret: {
+			Type:        schema.TypeString,
+			Description: descriptions.DatabaseSecretID,
+			Computed:    true,
+		},
+	}}
+
+func resourceDatabase() *schema.Resource {
+	return &schema.Resource{
+		Description:   descriptions.ResourceDatabase,
+		CreateContext: resourceDatabaseCreate,
+		ReadContext:   resourceDatabaseRead,
+		UpdateContext: resourceDatabaseUpdate,
+		DeleteContext: resourceDatabaseDelete,
+		Schema: map[string]*schema.Schema{
+			attributes.ID: {
+				Type:     schema.TypeString,
+				Computed: true,
+				// Description is autogenerated
+			},
+			attributes.ResourceGroup: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: descriptions.ResourceGroupID,
+			},
+			attributes.Project: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: descriptions.ProjectID,
+			},
+			attributes.CanonicalName: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: descriptions.CanonicalName,
+			},
+			attributes.DatabaseType: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: descriptions.DatabaseType,
+				ForceNew:    true,
+			},
+			attributes.RecipeBook: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: descriptions.RecipeBookID,
+			},
+			attributes.ManagementConnectionDetails: {
+				Type:        schema.TypeList,
+				Required:    true,
+				MinItems:    1,
+				MaxItems:    1,
+				Description: descriptions.ManagementConnectionDetails,
+				Elem:        dbConnectionTypes,
+				ForceNew:    true,
+			},
+			attributes.ManagementGatewaySelector: {
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Description: descriptions.ManagementGatewaySelector,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			attributes.ManagementGatewaySelectorID: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: descriptions.ManagementGatewaySelectorID,
+			},
+			attributes.CreatedAt: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: descriptions.CreatedAt,
+			},
+			attributes.UpdatedAt: {
```

Comment:
I don't think we generally expose this within the provider.  Not totally opposed to keeping it, but I don't think it's really that useful.

Reply:
This made me realize I wasn't actually setting them, nor `RecipeBook` - forgot those top level fields after focussing on the other problems.  I've removed those and now actually set the recipe.